proxy/proxy/client.go

```diff
@@ -195,13 +205,24 @@ func (c *Client) refreshCfg(instance string) (addr string, cfg *tls.Config, err
 		Certificates: []tls.Certificate{mycert},
 		RootCAs:      certs,
 	}
+
+	// Refresh cert 5 minutes before it expires.
+	timeToRefresh := cfg.Certificates[0].Leaf.NotAfter.Sub(time.Now()) - refreshCertBuffer
+	if timeToRefresh > 0 {
+		go func() {
+			<-time.After(timeToRefresh)
+			logging.Verbosef("Cert for instance %s will expire soon, refreshing now.", instance)
+			c.refreshCfg(instance)
```

Comment:
I think checking for an error here would be a good idea since it is still an error which users may care about. They'd also receive an error the next time they tried to dial, but I figure we might as well print it out now.

Reply:
Done.